src/figuresClasses.ts

```diff
@@ -1,11 +1,87 @@
-export interface Figure {}
+enum Shape {
+  'Triangle' = 'triangle',
+  'Circle' = 'circle',
+  'Rectangle' = 'rectangle',
+}
+
+type Color = 'red' | 'green' | 'blue';
+
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape = Shape.Triangle;
 
-export class Triangle implements Figure {}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const sorted: number[] = [a, b, c].sort((x: number, y: number) => x - y);
 
-export class Circle implements Figure {}
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw Error('Side length is 0 or less');
+    }
+
+    if (sorted[0] + sorted[1] <= sorted[2]) {
+      throw Error('Sides do not form a valid triangle');
+    }
+  }
+
+  getArea(): number {
```

Comment:
[CODE STYLE] - The `getArea` function doesn't need to accept any parameters. All necessary values are already available as object fields.